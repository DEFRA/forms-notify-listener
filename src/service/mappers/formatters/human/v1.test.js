import { buildDefinition } from '@defra/forms-model/stubs'

import {
  exampleNotifyFormDefinition,
  exampleNotifyFormMessage,
  pizzaFormDefinition,
  pizzaMessage
} from '~/src/service/mappers/formatters/__stubs__/notify.js'
import { getFormatter } from '~/src/service/mappers/formatters/index.js'

jest.mock('nunjucks', () => {
  const environment = {
    addFilter: jest.fn(),
    addGlobal: jest.fn()
  }
  return {
    configure: jest.fn(() => environment)
  }
})
jest.mock('~/src/config/index.js', () => ({
  config: {
    get: jest.fn(() => {
      return 'http://designer'
    })
  }
}))

describe('Page controller helpers', () => {
  beforeAll(() => {
    jest.useFakeTimers().setSystemTime(new Date('2025-09-01T00:00:00Z'))
  })
  afterAll(() => {
    jest.useRealTimers()
  })

  it('should return a valid human readable v1 response', () => {
    const definition = buildDefinition({
      ...exampleNotifyFormDefinition,
      output: {
        audience: 'human',
        version: '1'
      }
    })
    const formatter = getFormatter('human', '1')
    let output = formatter(exampleNotifyFormMessage, definition, '1')

    expect(output).toContain(
      '^ For security reasons, the links in this email expire at'
    )
    expect(output).toContain('Example notify form form received at')
    expect(output).toContain('## What is your name?')
    expect(output).toContain('Someone')
    expect(output).toContain('## Additional details')
    expect(output).toContain('## What is your address?')
    expect(output).toContain('1 Anywhere Street')
    expect(output).toContain('Anywhereville')
    expect(output).toContain('Anywhereshire')
    expect(output).toContain('AN1 2WH')
    expect(output).toContain('## What is your date of birth?')
    expect(output).toContain('1 January 2000')
    expect(output).toContain('## What month is it?')
    expect(output).toContain('August 2025')
    expect(output).toContain('## Who are your favourite LotR characters?')
    expect(output).toContain('* Gandalf')
    expect(output).toContain('* Frodo')
    expect(output).toContain('## Team Member')
    expect(output).toContain(
      '[Download Team Member \\(CSV\\)](http://designer/file-download/e3005cd2-8b1c-4dc4-b2ac-bd1ff73666a9)'
    )
    expect(output).toContain('## Please add supporting evidence')
    expect(output).toContain('Uploaded 1 file:')
    expect(output).toContain(
      '* [supporting\\_evidence\\.pdf](http://designer/file-download/ef4863e9-7e9e-40d0-8fea-cf34faf098cd)'
    )

    expect(output).toContain(
      '[Download main form \\(CSV\\)](http://designer/file-download/818d567d-ee05-4a7a-8c49-d5c54fb09b16)'
    )
    output = output.replace(/(12|1):00am/g, '12:00am')
    expect(output).toMatchSnapshot()
  })

  it('should return a valid human readable v1 response 2', () => {
    const definition = buildDefinition({
      ...pizzaFormDefinition,
      output: {
        audience: 'human',
        version: '1'
      }
    })
    const formatter = getFormatter('human', '1')
    let output = formatter(pizzaMessage, definition, '1')
    output = output.replace(/(12|1):00am/g, '12:00am')
    expect(output).toMatchSnapshot()
  })
})
