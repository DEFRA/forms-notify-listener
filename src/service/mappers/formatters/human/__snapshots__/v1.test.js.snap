// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Page controller helpers should return a valid human readable v1 response 1`] = `
"^ For security reasons, the links in this email expire at 12:00am on Sunday 30 November 2025

Example notify form form received at 12:00am on 1 September 2025.

---

## What is your name?

Someone

---

## Additional details

---

## What is your address?

1 Anywhere Street
Anywhereville
Anywhereshire
AN1 2WH

---

## What is your date of birth?

1 January 2000

---

## What month is it?

August 2025

---

## Team Member

[Download Team Member \\(CSV\\)](http://designer/file-download/e3005cd2-8b1c-4dc4-b2ac-bd1ff73666a9)

---

## Who are your favourite LotR characters?

* Gandalf
* Frodo

---

## Please add supporting evidence

Uploaded 1 file:

* [supporting\\_evidence\\.pdf](http://designer/file-download/ef4863e9-7e9e-40d0-8fea-cf34faf098cd)

---

[Download main form \\(CSV\\)](http://designer/file-download/818d567d-ee05-4a7a-8c49-d5c54fb09b16)
"
`;

exports[`Page controller helpers should return a valid human readable v1 response 2 1`] = `
"^ For security reasons, the links in this email expire at 1:00am on Sunday 30 November 2025

Pizza Form form received at 1:00am on 1 September 2025.

---

## Pizza

[Download Pizza \\(CSV\\)](http://designer/file-download/11111111-1111-1111-1111-111111111111)

---

## How would you like to receive your pizza?

Delivery

---

[Download main form \\(CSV\\)](http://designer/file-download/00000000-0000-0000-0000-000000000000)
"
`;
